src/app/zap-templates/templates/app/cluster-objects.zapt

```diff
@@ -71,11 +74,44 @@ namespace {{name}} {
 namespace Commands {
 {{/first}}
 namespace {{asUpperCamelCase name}} {
-struct Type {};
+enum FieldId {
+    {{#zcl_command_arguments}}
+    k{{asUpperCamelCase label}}FieldId = {{index}},
+    {{/zcl_command_arguments}}
+};
 
-struct DecodableType {};
-} // namespace {{asUpperCamelCase name}}
+struct Type
+{
+public:
+    static constexpr chip::CommandId GetCommandId() { return {{asUpperCamelCase name}}::Id; }
+    static constexpr chip::ClusterId GetClusterId() { return {{asUpperCamelCase parent.name}}::Id; }
```

Comment:
```suggestion
    static constexpr CommandId GetCommandId() { return {{asUpperCamelCase name}}::Id; }
    static constexpr ClusterId GetClusterId() { return {{asUpperCamelCase parent.name}}::Id; }
```

Reply:
Why have it be funcs? Feels like its more consistent to just have it as IDs:

```
static constexpr CommandId CommandId = {{asUpperCamelCase name}}::Id;
static constexpr ClusterId ClusterId = {{asUpperCamelCase parent.name}}::Id;
```

Reply:
in identification cluster, there is argument called commandid, so for static constextpr command id, I add prefix s.
static constexpr chip::CommandId commandId = ExecutionOfACommand::Id;
// static constexpr chip::ClusterId clusterId = ::Id;
CommandIdentification commandId; ^~

https://github.com/project-chip/connectedhomeip/pull/10171#discussion_r722834596
Boris suggest static constexpr GetCommandId()